src/clj_kondo/impl/linters/config.clj

```diff
@@ -47,6 +48,58 @@
                      :clj-kondo-config
                      (str "Linter config should go under :linters")))))))
 
+(defn ^:private minimum-version-node
```

Comment:
I would move the processing of `:min-clj-kondo-version` to just the `impl/config.clj` namespace or so, or even `impl/core.clj`. This namespace is about verifying the `config.edn` file for syntactical errors.

Reply:
One more reason to not have the check here is that configurations might be merged and version can be overridden. As such we should only look at the effective config and not at a single config file.

Reply:
That's reasonable - I didn't realise that the linting here would apply only to the file.

This does raise a question though of how to register findings  not associated with a file. When registering a finding, we have to give a map containing `:filename`, `:row` and `:col`. `:row` and `:col` might make sense from configuration from any source, but `:filename` may not, e.g. a configuration given on the command line. I'm also not sure that it will be clear to this code where the `:min-clj-kondo-version` came from, i.e. was it a `config.edn`, was it from the command line, etc... Any thoughts on how to handle such a case?

Reply:
That's a good point! Maybe this tips the balance in favor of just printing to stderr?

Reply:
Yes, thinking about it a bit more, this would also require (probably substantial) changes to config reading to track provenance (e.g. to know if the minimum version came from a file, from the command line, and in which row/col of the appropriate one) as well as changes to interfaces around findings - probably not worth it for this feature.

Also good to note that it's not too difficult to reverse this decision in future if `clj-kondo` changes in such a way that it makes more sense to have this check as a finding.

I've drafted up this approach in 9e5e395c897e9b67e2c6d187be3eed654ea4e89e & 475ce63b92060e428571eb1f657b2e51812c2424 - let me know what you think!

Reply:
I'll have to look later this week into the test failures. It passes locally, I haven't yet figured out why it fails in CI.